main.cpp

```diff
@@ -56,48 +56,80 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value; //2
+    std::string name; //3
+    T(int v, std::string n) : value(v), name(n) {}//1
 };
 
-struct <#structName1#>                                //4
+struct CheckPointer                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if ( ( a != nullptr ) && ( b != nullptr) )
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
         return nullptr;
```

Comment:
You need to add curly braces to make this `if()` statement apply to both of these internal `if()` statements that return values when their condition is true.
When you omit the curly braces, only the first line is executed when the condition is true:
```
if( condition )
    expression //only executed when 'condition' is true.
    expression //this is ALWAYS executed, regardless of condition. 
``` 
vs: 
```
if( condition )
{
    expression1 //only executed if condition is true
    expression2 //only executed if condition is true
}
```
- [ ]revise